lib/collection/src/shards/local_shard/clock_map.rs

```diff
@@ -321,15 +350,26 @@ impl From<RecoveryPoint> for api::grpc::qdrant::RecoveryPoint {
     }
 }
 
-impl From<api::grpc::qdrant::RecoveryPoint> for RecoveryPoint {
-    fn from(rp: api::grpc::qdrant::RecoveryPoint) -> Self {
+impl TryFrom<api::grpc::qdrant::RecoveryPoint> for RecoveryPoint {
+    type Error = Status;
+
+    fn try_from(rp: api::grpc::qdrant::RecoveryPoint) -> Result<Self, Self::Error> {
         let clocks = rp
             .clocks
             .into_iter()
-            .map(|tag| (Key::new(tag.peer_id, tag.clock_id), tag.clock_tick))
-            .collect();
+            .map(|tag| {
+                tag.token
+                    .try_into()
+                    .map_err(|err| {
+                        Status::invalid_argument(format!("Malformed clock tag token: {err}"))
+                    })
+                    .map(|token: Uuid| {
+                        (Key::new(tag.peer_id, tag.clock_id), (tag.clock_tick, token))
+                    })
+            })
+            .collect::<Result<_, _>>()?;
 
-        Self { clocks }
+        Ok(Self { clocks })
```

Comment:
I'd _maybe_ rewrite with an explicit loop. It's super ugly and hard to read. 😅

This is better, IMO.
```rust
let mut clocks = HashMap::new();

for tag in rp.clocks {
    let token = tag.token.try_into().map_err(|err| {
        Status::invalid_argument(format!("Malformed clock tag token: {err}"))
    })?;

    clocks.insert(Key::new(tag.peer_id, tag.clock_id), (tag.clock_tick, token));
}

Ok(Self { clocks })
```

Reply:
Sure!

Does `HashMap::with_capacity(...)` make sense here?

Reply:
Yeah, absolutely!